File: Desktop/Domain/CmdMsgs/SMSCmd.cs
using System;

using System.Xml;
using System.Xml.Xsl;
using System.Collections;
using System.Runtime.InteropServices;

namespace Blurts
{
  [ComVisible(true)]
  public class SMSCmd : CmdBase
  {
    // Tag Names
    private static String KEY_ADDRESS = "SenderAddress";
    private static String KEY_BODY = "BodyText";


    public SMSCmd(String s)
      : base(s)
    {
    }

    public SMSCmd()
      : base(CMD_SENDSMS)
    {
    }

    [ComVisible(true)]
    public String Address
    {
      get
      {
        return optString(KEY_ADDRESS);
      }
      set
      {
        try
        {
          remove(KEY_ADDRESS);
          put(KEY_ADDRESS, value);
        }
        catch (JSONException e)
        {
          Console.WriteLine(e.ToString());
        }
      }
    }

    [ComVisible(true)]
    public String Text
    {
      get
      {
        return optString(KEY_BODY);
      }
      set
      {
        try
        {
          remove(KEY_BODY);
          put(KEY_BODY, value);
        }
        catch (JSONException e)
        {
          Console.WriteLine(e.ToString());
        }
      }
    }
  }
}

File: BlackBerry/AlertMsgs/SMSCmd.java
package com.mlhsoftware.Blurts;

import org.json.me.*;


class SMSCmd extends CmdBase 
{
  // Tag Names
  private static final String KEY_ADDRESS = "SenderAddress";
  private static final String KEY_BODY = "BodyText";


  public SMSCmd( String string ) throws JSONException
  {
    super( string );
  }

  public SMSCmd()
  {
    super( CMD_SENDSMS );
  }

  String getAddress()
  {
    return optString( KEY_ADDRESS );
  }

  String getText()
  {
    return optString( KEY_BODY );
  }
}
